lib/msf/ui/console/module_command_dispatcher.rb

```diff
@@ -173,7 +173,7 @@ def cmd_check_help
   def report_vuln(instance)
     framework.db.report_vuln(
       workspace: instance.workspace,
-      host: instance.rhost,
+      host: instance.datastore['RHOST'],
```

Comment:
I don't think this is the right long term solution.

`rhost` is expected to be the singular target the instance is run against, while `datastore['RHOST']` may or may not be set by the user.  Usage of `datastore['RHOSTS']` provides some complexity to how these values are set.  The fact that this change solves the inconsistency in the external module reporting suggest that there is a bug somewhere else in the logic that should be setting `rhost`.

Reply:
Note #17792 may also have had impacts to this flow.

Reply:
> Hey @micahjmartin, are you still investigating this issue? If so I'll leave this PR open for further updates, if not I'll close it. No worries either way, just let me know, cheers.

Hi sorry, I am not, after seeing that it might be a bigger change than what I originally proposed, I stopped looking. I can dig into it more though. Ill see if I can figure it out, if not ill close it